registration-processor/post-processor/registration-processor-credential-requestor-stage/src/main/java/io/mosip/registration/processor/credentialrequestor/util/CredentialPartnerUtil.java

```diff
@@ -75,35 +92,50 @@ public List<String> getCredentialPartners(String regId, String registrationType,
             regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(),
                     LoggerFileConstant.REGISTRATIONID.toString(), regId,
                     PlatformErrorMessages.RPR_PRT_ISSUER_NOT_FOUND_IN_PROPERTY.name());
-            return filteredPartners;
+            return Lists.emptyList();
         }
 
+        String schemaVersion = utilities.getPacketManagerService().getFieldByMappingJsonKey(regId,
+                MappingJsonConstants.IDSCHEMA_VERSION, registrationType, ProviderStageName.CREDENTIAL_REQUESTOR);
+
         Map<String, String> identityFieldValueMap = utilities.getPacketManagerService().getFields(regId,
-                requiredIdObjectFieldNames, registrationType, ProviderStageName.EVENT_HANDLER);
+                idSchemaUtil.getDefaultFields(Double.valueOf(schemaVersion)), registrationType, ProviderStageName.CREDENTIAL_REQUESTOR);
```

Comment:
Any specific reason to get all the default fields rather than requiredIdObjectFieldNames ? This was implemented to avoid getting all the fields and looping through everything has a performance impact.

Reply:
Even the earlier code with requiredIdObjectFieldNames method will read the mvel expression, take any field that is configured in the mvel expression and fetch only those fields from packet manager.